packages/worker/src/spot-termination.ts

```diff
@@ -0,0 +1,35 @@
+import {
+	changeMessageVisibility,
+	logger,
+} from '@guardian/transcription-service-backend-common';
+import { SQSClient } from '@aws-sdk/client-sqs';
+import { getCurrentReceiptHandle, setInterruptionTime } from './index';
+
+const CHECK_FREQUENCY = 10;
+
+export const checkSpotInterrupt = async (
+	client: SQSClient,
+	queueUrl: string,
+) => {
+	const url = 'http://169.254.169.254/latest/meta-data/spot/instance-action';
+	const result = await fetch(url);
+	if (result.status === 200) {
+		const json = await result.json();
+		if (json.action === 'terminate') {
+			setInterruptionTime(new Date(json.time));
+			logger.warn('Spot instance termination detected');
+			// Interrupt warning occurs 2 minutes before termination
+			const receiptHandle = getCurrentReceiptHandle();
+			if (!receiptHandle) {
+				return;
+			}
+			await changeMessageVisibility(client, queueUrl, receiptHandle, 110);
```

Comment:
Might be worth considering relying on termination time from instance metadata rather than the two minutes from docs. this way message visibility and `INTERRUPTION_TIME` agree
```suggestion
			await changeMessageVisibility(client, queueUrl, receiptHandle, new Date(json.time) - new Date());
```